### PR TITLE
Start auto dasker combine

### PR DESCRIPTION
**Pull request by martindurant**

### CODE REVIEW

**Comment by martindurant** [comment]
Still needs: 
- more complicated test cases
- own documentation section, with reference to "by hand" version like https://gist.github.com/peterm790/5f901453ed7ac75ac28ed21a7138dcf8

**Comment by martindurant** [comment]
@peterm790 , if you have enthusiasm and time, would be interesting to see whether the ERA5 workflow can be turned into a one-lines using this function. Would probably take some trial-and-error to get right.

File: kerchunk/combine.py [hunk 1]
```diff
@@ -578,3 +590,90 @@ def concatenate_arrays(
...
     return consolidate(out)
+
+
+def auto_dask(
+    urls: List[str],
+    single_driver: str,
+    single_kwargs: dict,
+    mzz_kwargs: dict,
+    n_batches: int,
+    remote_ptotocol=None,
```
Comment (peterm790) [inline kerchunk/combine.py:601]:
```suggestion
    remote_protocol=None,
```

File: kerchunk/combine.py [hunk 2]
```diff
@@ -578,3 +590,90 @@ def concatenate_arrays(
...
+    Parameters
+    ----------
+    urls: list of input datasets
+    single_driver: class with ``translate()`` method
+    single_kwargs: to pass to single-input driver
+    mzz_kwargs: passed to ``MultiZarrToZarr`` for each batch
+    n_batches: int
+        Number of MZZ instances in the first combine stage. Maybe set equal
+        to the number of dask workers, or a multple thereof.
+    remote_ptotocol: str | None
```
Comment (peterm790) [inline kerchunk/combine.py:620]:
```suggestion
    remote_protocol: str | None
```

**Review by peterm790 — COMMENTED** [review]
This seems to be working well for me. 

I had to make a small change in SingleHdf5ToZarr to open the files from within kerchunk I have made a seperate PR for that https://github.com/fsspec/kerchunk/pull/256

Here is the old ERA5 example using `auto_dask` : https://gist.github.com/peterm790/378c48ae61a423d050b24abf772f0d95

**Pull request merged**